### PR TITLE
Merge unee t ins improvements

### PR DESCRIPTION
**Pull request by franck-boullier**
This is to simplify the codebase and make sure that we can have one single repo for this dependancy for any Unee-T installation.

WARNING:
In each of the environments that will need this code, the following AWS secrets MUST have been declared:
- DOMAIN
- INSTALLATION_ID
- STAGE
- DEFAULT_REGION
- MYSQL_HOST
- BUGZILLA_DB_USER
- BUGZILLA_DB_PASSWORD

@kaihendry can you have a look at this please?

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -119,12 +122,16 @@ func (e Env) BugzillaDSN() string {
...
+	}
+
+	if mysqlhost == "" {
+		log.Fatal("MYSQL_HOST is unset")
 	}
 
 	return fmt.Sprintf("%s:%s@tcp(%s:3306)/bugzilla?multiStatements=true&sql_mode=TRADITIONAL&timeout=5s&collation=utf8mb4_unicode_520_ci",
-		e.GetSecret("MYSQL_USER"),
-		e.GetSecret("MYSQL_PASSWORD"),
+		e.GetSecret("BUGZILLA_DB_USER"),
```
Comment (kaihendry) [inline main.go:142]: I'm not sure what you are doing here. Are you trying to backport the changes in the fork uneet-ins/env here?

Since in the uneet-ins environment you have changed variable names and they are not updated in:

```
[hendry@t480s ~]$ ssm uneet-prod BUGZILLA_DB_PASSWORD
None
[hendry@t480s ~]$ ssm uneet-demo BUGZILLA_DB_PASSWORD
None
```

File: README.md [hunk 2]
```diff
@@ -1 +1,61 @@
...
+)
+```
+
+- Deploy the updated code for the relevant Unee-T module (this is usually done via a tag release of the code in the repo for that code).
+
+## Option 2 - Update main.mod file
+
+- In the codebase for the relevant Unee-T module, update the relevant line in file `main.mod` to the new [version] number :
+Example:
+```
```
Comment (kaihendry) [inline README.md]: The way dependencies are handled, is via the Go toolchain. You don't edit go.mod by hand.

Go modules is relatively new so some repos might not be updated to use them.

File: main.go (hunk 1 above)
Comment (franck-boullier) [reply main.go:142]: > 'm not sure what you are doing here. Are you trying to backport the changes in the fork uneet-ins/env here?

Yes that's what I'm trying to do here

Comment (kaihendry) [reply main.go:142]: Why? You have changed names and backporting these name changes will create work.

Comment (franck-boullier) [reply main.go:142]: > Since in the uneet-ins environment you have changed variable names and they are not updated in:
```
[hendry@t480s ~]$ ssm uneet-prod BUGZILLA_DB_PASSWORD
None
[hendry@t480s ~]$ ssm uneet-demo BUGZILLA_DB_PASSWORD
None
```
These have been updated in the PROD environment now.
Will do the DEMO is a few minutes

File: README.md (hunk 2 above)
Comment (franck-boullier) [reply README.md]: > You don't edit go.mod by hand.

Can you rephrase the doc in the Readme then so it reflects the reality

File: main.go (hunk 1 above)
Comment (franck-boullier) [reply main.go:142]: The parameters BUGZILLA_DB_PASSWORD and BUGZILLA_DB_USER have been added to the parameters store for the DEMO and PROD environment now

Comment (kaihendry) [reply main.go:142]: So is the idea now that uneet-ins projects use unee-t/env since now it's the same as Unee-T-INS/env?

The reason why people use libraries and such is reduce code duplication. Having two repos doing the exact same thing creates confusion.

Comment (franck-boullier) [reply main.go:142]: > So is the idea now that uneet-ins projects use unee-t/env since now it's the same as Unee-T-INS/env?

Yes, this is exactly the idea.

Comment (franck-boullier) [reply main.go:142]: Once we are comfortable that unee-t/env can work nicely with both Unee-T and any other Unee-T installation (including but not limited to Unee-T INS) then we will have no need for the `uneet-ins/env` fork in the first place.

Ideally we can do the same for all dependencies (apienroll, invite, unit, lambda2sqs, etc...)

File: main.go [hunk 3]
```diff
@@ -76,13 +79,17 @@ func (e Env) Bucket(svc string) string {
...
+		log.Warnf("Using fallback INSTALLATION_ID: %s: ", installationID)
+	}
+	if installationID == "main" {
+		// Preserve original bucket names
+		return fmt.Sprintf("%s-%s-unee-t", e.Stage, svc)
+	} else {
+		// Use INSTALLATION_ID to generate unique bucket name
+		return fmt.Sprintf("%s-%s-%s", e.Stage, svc, installationID)
 	}
 }
```
Comment (franck-boullier) [inline main.go:95]: @kaihendry the variable `region` looks undefined I think it should be `defaultRegion` instead.
See line 96 and 101.

`func (e Env) SNS(name, region string) string {`

`return fmt.Sprintf("arn:aws:sns:%s:%s:%s", region, e.AccountID, name)`

Comment (kaihendry) [reply main.go:95]: Region is set by arguments passed to the function. It differs in some cases since some things are only present in certain Regions.
